src/test/java/com/wikia/webdriver/testcases/infoboxbuilder/InfoboxBuilderTests.java

```diff
@@ -160,31 +179,59 @@ public void verifyInfoboxPreviewTheme() {
   preview is greater than the preview area height. */
   @Execute(asUser = User.USER)
   public void verifyScrolling() {
-    new InfoboxBuilderPage().open("verifyScrolling")
+    new InfoboxBuilderPage().open("InfoboxBuilderVerifyScrolling")
         .switchToIFrame()
         .addImageComponent()
         .addImageComponent()
         .addImageComponent()
         .addImageComponent()
         .verifyScrollbarIsVisible()
-        .scrollAndSelectLastComponent();
+        .selectLastComponent();
   }
 
   @Execute(asUser = User.STAFF)
   public void verifyUserInteractions() {
-    new InfoboxBuilderPage().open("verifySelectedBorderStyling")
+    InfoboxBuilderPage builder = new InfoboxBuilderPage();
+
+    builder.open("InfoboxBuilderVerifySelectedBorderStyling")
         .switchToIFrame()
-        .verifySelectedComponentBorderStyle(0)
         .verifyTooltipOnHover();
+
+    String borderStyle = builder.getBorderStyle();
+    Assertion.assertEquals(borderStyle, "1px solid rgb(26, 94, 184)");
+
+    borderStyle = builder.clickBuilderBackground().getBackgroundColor();
+    Assertion.assertNotEquals(borderStyle, "1px solid rgb(26, 94, 184)");
   }
 
   @Execute(asUser = User.STAFF)
   public void verifyReordering() {
-    new InfoboxBuilderPage().open("verifyReordering")
+    new InfoboxBuilderPage().open("InfoboxBuilderVerifyReordering")
         .switchToIFrame()
         .dragAndDropToTheTop(2)
         .dragAndDropToTheTop(3)
         .dragAndDropToTheTop(1);
   }
 
+  @Execute(asUser = User.USER)
+  public void verifyRedirectingUnsupportedInfoboxes() {
+    new InfoboxBuilderPage().open("InfoboxBuilderMultipleInfoboxes");
+    Assertion.assertTrue(new TemplateEditPage().isEditAreaDisplayed());
+
+    new InfoboxBuilderPage().open("InfoboxBuilderUnsupportedMarkup");
+    Assertion.assertTrue(new TemplateEditPage().isEditAreaDisplayed());
+  }
+
+  @Execute(asUser = User.USER)
+  public void immutableSources() {
+    new InfoboxBuilderPage().open("InfoboxBuilderImmutableRows")
+        .switchToIFrame()
```

Comment:
That's actually a good point. I moved it inside ` open()` method and it will return opened, ready-for-interaction on iFrame object.